public/src/components/switchboard.tsx

```diff
@@ -97,61 +136,164 @@ const Switchboard: React.FC<InnerProps<SupportFrontendSwitches>> = ({
     );
   };
 
+  const setPendingChange = (changeName: string, description: string, groupId: string): void => {
+    const unsavedChange = changeName + ' ' + description + ' in ' + groupId;
+    // Add the changeName to the list of pendingChanges
+    setPendingChanges(prevChanges => [...prevChanges, unsavedChange]);
+  };
+
   const updateSwitchSetting = (
     switchId: string,
     switchData: Switch,
-    groupId: string,
+    group: [string, SwitchGroup],
     isChecked: boolean,
   ): void => {
     const updatedState = cloneDeep(data);
-
+    const [groupId, groupData] = group;
     updatedState[groupId].switches[switchId].state = isChecked ? SwitchState.On : SwitchState.Off;
-
+    const currentSwitchState = updatedState[groupId].switches[switchId].state;
     setData(updatedState);
+    setNewUnsavedData(true);
+    setPendingChange(
+      'Turned ' + currentSwitchState + ':',
+      switchData.description,
+      groupData.description,
+    );
     setNeedToSaveDataWarning(true);
   };
 
   const createSwitchesFromGroupData = (
     switchId: string,
     switchData: Switch,
-    groupId: string,
+    group: [string, SwitchGroup],
   ): JSX.Element => {
     const isChecked = switchData.state === 'On';
 
     return (
-      <FormControlLabel
-        label={switchData.description}
-        checked={switchData.state === 'On'}
-        control={<SwitchUI />}
-        onChange={(): void => updateSwitchSetting(switchId, switchData, groupId, !isChecked)}
-        value={switchId}
-        key={switchId}
-      />
+      <div className={classes.switchLayout}>
+        <FormControlLabel
+          label={switchData.description}
+          checked={switchData.state === 'On'}
+          control={<SwitchUI />}
+          onChange={(): void => updateSwitchSetting(switchId, switchData, group, !isChecked)}
+          value={switchId}
+          key={switchId}
+        />
+        <IconButton
+          onClick={(): void => actionRemoveSwitchData(group, switchId, switchData.description)}
+          aria-label="removeSwitch"
+        >
+          <DeleteIcon />
+        </IconButton>
+      </div>
     );
   };
 
   const createGroupsFromData = (group: [string, SwitchGroup]): JSX.Element => {
     const [groupId, groupData] = group;
+    const { register, handleSubmit, errors } = useForm<FormData>();
+
+    const onSubmit = ({ switchId, description }: FormData): void => {
+      const updatedState = cloneDeep(data);
+
+      updatedState[groupId].switches[switchId] = {
+        description: description,
+        state: SwitchState.Off,
+      };
+      setData(updatedState);
+      setNewUnsavedData(true);
+      setPendingChange('Added', description, groupData.description);
+      setNeedToSaveDataWarning(true);
+    };
 
     return (
       <FormControl className={classes.formControl} key={groupId}>
-        <FormLabel>{groupData.description}</FormLabel>
+        <FormLabel>
+          {' '}
+          <strong>{groupData.description} </strong>
```

Comment:
Are these spaces meant to be there?

```suggestion
          <strong>{groupData.description}</strong>
```